packages/searchbase/src/SearchComponent.js

```diff
@@ -798,9 +798,10 @@ class SearchComponent extends Base {
   // Method to execute the component's own query i.e default query
   triggerDefaultQuery = (options?: Option = defaultOption): Promise<any> => {
     // To prevent duplicate queries
-    if (isEqual(this._query, this.componentQuery)) {
+    if (isEqual(this._query, JSON.parse(JSON.stringify(this.componentQuery)))) {
```

Comment:
What could be the reason of changing it?

Reply:
when the `_query` is hydrated from the serverside, it doesn't contain any key-value pair with a value as `undefined` since undefined is not serializable

now, the `componentQuery` can contain key-value pair with value `undefined` and thus we need to somehow get rid of these keys with values as `undefined`.

eg: **_query = { a: some}** is not compared correctly with **componentQuery = {a: some, b: undefined}**

**JSON.parse(JSON.stringify(this.componentQuery))** removes any key-value pair with value as `undefined`

```
let componentQuery = {a: some, b: undefined};

componentQuery = JSON.parse(JSON.stringify(componentQuery));

console.log(componentQuery); // gives out as  {a: some}
```

Reply:
Ok, use the same logic for `this._query` too.